ext/console/02_console.js

```diff
@@ -805,24 +829,46 @@
     inspectOptions,
   ) {
     const gray = maybeColor(colors.gray, inspectOptions);
+    let lastValidIndex = 0;
+    let keys;
     const options = {
       typeName: "Array",
       displayName: "",
       delims: ["[", "]"],
-      entryHandler: (entry, inspectOptions, next) => {
+      entryHandler: (entry, inspectOptions) => {
         const [index, val] = entry;
         let i = index;
+        lastValidIndex = index;
         if (!ObjectPrototypeHasOwnProperty(value, i)) {
-          i++;
-          while (!ObjectPrototypeHasOwnProperty(value, i) && i < value.length) {
-            next();
-            i++;
+          let skipTo;
+          keys = keys || ObjectKeys(value);
+          if (keys.length === 0) {
+            // fast path, all items are empty
+            i = value.length;
+            skipTo = i;
+          } else {
+            // Not all indexes are empty or there's a non-index property
+            // Find first non-empty array index
+            i = value.length;
```

Comment:
`i = value.length` occurs either way, hoist it out?

Reply:
Good catch, was too tired when cleaning out the code and forgot a few things :)

Reply:
And one more comment - now that it's hoisted out, it's still a polymorphic access, so should be `entriesLength`?